File: fitcore-backend/Entity/Meal.cs
namespace fitcore_backend.Entity;

/// <summary>
/// Прием пищи
/// </summary>
public class Meal
{
    public Guid Id { get; set; }
    public Guid NutritionDayId { get; set; }

    public MealType MealType { get; set; }

    public double Protein { get; set; }
    public double Fat { get; set; }
    public double Carbs { get; set; }

    public NutritionDay NutritionDay { get; set; } = null!;
}

public enum MealType {
    Breakfast = 0,
    Lunch = 1,
    Dinner = 2,
    Snacks = 3
}

File: fitcore-frontend/src/models/Meal.ts
export enum MealType {
    Breakfast = 0,
    Lunch = 1,
    Dinner = 2,
    Snacks = 3
}

export interface Meal {
    id: string;                // Идентификатор приема пищи
    nutritionDayId: string;    // День питания
    mealType: MealType;            // Тип приема пищи
    protein: number;           // Белки
    fat: number;               // Жиры
    carbs: number;             // Углеводы
}
